src/_zkapauthorizer/replicate.py

```diff
@@ -244,17 +261,36 @@ def snapshot(self) -> bytes:
         """
         return snapshot(self._conn)
 
+    # XXX there is a "commit" method too?
+
     def close(self):
         return self._conn.close()
 
     def __enter__(self):
         return self._conn.__enter__()
 
     def __exit__(self, *args):
-        return self._conn.__exit__(*args)
+        r = self._conn.__exit__(*args)
+        if self._replicating and r != (None, None, None):
```

Comment:
while merging main I had to make some checks here to satisfy mypy, so I just made what I think were the correct changes.  There are still some missing tests to write - the previous version should have been turning something red, probably to do with exception propagation through the context manager.

Reply:
hmm, was `__exit__` before just failing to run entirely because it lacked `**kwargs`...?

Reply:
From discussion elsewhere, the conclusion here is that the wrapped `__exit__` always returns False because the `sqlite3.Connection` context manager never suppresses exceptions.